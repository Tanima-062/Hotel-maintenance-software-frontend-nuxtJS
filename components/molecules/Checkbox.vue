<template>
  <div class="relative">
    <div
      v-show="showTooltip"
      class="tooltip-position shadow break-all bg-bgColor-light-gray w-70 rounded"
    >
      {{ tooltip }}
    </div>
    <label
      class="flex items-center focus:outline-none"
      :class="styles"
      tabindex="-1"
      @mouseover="mouseover"
      @mouseleave="mouseleave"
    >
      <input
        type="checkbox"
        tabindex="0"
        class="form-checkbox focus:shadow-outline focus:outline-none"
        :class="disabled && 'text-disabled'"
        :checked="checked"
        :disabled="disabled"
        @change="change"
      />
      <span class="ml-2" :class="`text-${checked ? 'checkboxColor' : color}`">{{
        label
      }}</span>
    </label>
  </div>
</template>
<script lang="ts">
import { Vue, Component, Prop, Emit } from 'vue-property-decorator'

@Component
export default class extends Vue {
  @Prop({ default: false })
  checked!: boolean

  @Prop({ default: '' })
  label!: string

  @Prop({ default: 'black' })
  color!: string

  @Prop({ default: false })
  disabled!: boolean

  @Prop({ default: '' })
  tooltip!: string

  private isShowTooltip: boolean = false

  get styles(): string {
    return this.checked ? 'text-checkboxColor' : 'text-black'
  }

  get showTooltip(): boolean {
    return this.isShowTooltip && this.tooltip !== ''
  }

  @Emit()
  private change(e: Event & { target: HTMLInputElement }): boolean {
    return e.target.checked
  }

  private mouseover(): void {
    this.isShowTooltip = true
  }

  private mouseleave(): void {
    this.isShowTooltip = false
  }
}
</script>
<style lang="sass" scoped>
.tooltip-position
  position: absolute
  bottom: 110%
  left: 5%
</style>
