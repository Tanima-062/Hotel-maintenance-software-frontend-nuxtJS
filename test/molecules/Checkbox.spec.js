import { mount } from '@vue/test-utils'
import Checkbox from '@/components/molecules/Checkbox.vue'

describe('Molecules Checkbox Check', () => {
  test('Checkbox Test', () => {
    const wrapper = mount(Checkbox)
    // check if it is a vue element
    expect(wrapper.vm).toBeTruthy()

    // check Snapshot
    expect(wrapper.element).toMatchSnapshot()
  })
})
