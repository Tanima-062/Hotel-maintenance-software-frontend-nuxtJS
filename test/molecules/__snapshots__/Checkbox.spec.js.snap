// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Molecules Checkbox Check Checkbox Test 1`] = `
<div
  class="relative"
>
  <div
    class="tooltip-position shadow break-all bg-bgColor-light-gray w-70 rounded"
    style="display: none;"
  >
    
    
  
  </div>
   
  <label
    class="flex items-center focus:outline-none text-black"
    tabindex="-1"
  >
    <input
      class="form-checkbox focus:shadow-outline focus:outline-none"
      tabindex="0"
      type="checkbox"
    />
     
    <span
      class="ml-2 text-black"
    >
      
    </span>
  </label>
</div>
`;
